sdk/cosmosdb/cosmos/src/request/request.ts

```diff
@@ -127,6 +127,11 @@ export async function getHeaders({
     headers[Constants.HttpHeaders.ConsistencyLevel] = options.consistencyLevel;
   }
 
+  if (options.maxIntegratedCacheStaleness) {
```

Comment:
From offline sync: Only set this header when the resourceType == document/item

Reply:
fixed